src/applications/personalization/profile-2/msw-mocks.js

```diff
@@ -0,0 +1,62 @@
+import { rest } from 'msw';
+import environment from 'platform/utilities/environment';
+
+export const newPaymentAccount = {
+  accountType: 'Savings',
+  financialInstitutionName: 'COMERICA BANK',
+  accountNumber: '*****6789',
+  financialInstitutionRoutingNumber: '*****4321',
+};
+
+const prefix = environment.API_URL;
+
+export const updateDirectDepositSuccess = [
+  rest.put(
+    // I'd prefer to just set the route as `ppiu/payment_information` or at least `v0/ppiu/payment_information`
+    `${prefix}/v0/ppiu/payment_information`,
+    (req, res, ctx) => {
+      return res(
+        ctx.json({
+          data: {
+            attributes: {
+              responses: [
+                {
+                  paymentAccount: newPaymentAccount,
+                },
+              ],
+            },
+          },
+        }),
+      );
+    },
+  ),
+];
+
+export const updateDirectDepositFailure = [
+  rest.put(`${prefix}/v0/ppiu/payment_information`, (req, res, ctx) => {
```

Comment:
I'm curious why you added `msw` over using the fetch mocking we do now. It looks like more code for a benefit that's unclear to me.

Reply:
The honest answer is because I didn't think we had an existing method for mocking specific endpoints like this. 😬  I had checked with Nick who also wasn't aware of us doing this in our codebase but he told me about `msw`. It wasn't until I was trying to get my PR to pass Jenkins that I noticed you using `withArgs()` in some of your recent VAOS tests. 🤦  I'm happy to remove `msw` and do it how you've done it with VAOS.

But some possible benefits of using `msw`:
- I do like the Express-like syntax of mocking endpoints with `msw`.
- `msw` would also make it easy to provide different mocks for different methods of the same endpoint (obviously not something we need now)
- I would like to use `msw` in place of our hacky local mocking of the VA Profile and connected apps endpoints as well, but that would be further down the road

Reply:
Ah, ok. I'm not trying to suggest you switch back to the fetch mocks, I was just curious if there were benefits I was missing.

Reply:
I think it plays really well with React Testing Library in that you don't need to mock fetch in a component because the response will be served invisibly by MSW, so you can just check the component for the loading then render state. It's neat thinking of testing the component at the user/visual level, where maybe `fetch` itself is an implementation detail

Reply:
MSW is new therefore it must be cool 😎 

Actually the fact that it's so new and not widely used yet gives me pause.

Reply:
I think in both cases the tests in RTL are basically the same. You just wait for the right stuff on your page to show up. But with MSW you're actually running the fetch library code, too. That could be useful, but we're in Node and running node-fetch anyway, rather than a browser implementation.

Sharing mocks across tests and with local development could be a compelling feature. You'd have one tool for both, instead of our custom fetch mocking and custom local mock api.

Reply:
Agreed that we still have to set up the API mocks for a particular test and it makes little difference if we do that using our Sinon stub or `msw`. The reason I'd lean towards keeping this addition of `msw` is (1) the nicer API (2) we could potentially do other cool stuff with `msw` in the future.